### PR TITLE
NAS-117470 / 23.10 / Query formatted disks gptids after mass formatting to prevent race co…

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
time 0:30

**Review by yocalebo — CHANGES_REQUESTED** [review]
1. update docstring
2. `system.is_ha_capable` can be called outside the inner `format_disk` method and simply passed as another arg
3. call `len(disks)` outside inner `format_disk` method and pass it in as an arg too

File: src/middlewared/middlewared/plugins/pool_/format_disks.py [hunk 1]
```diff
@@ -7,30 +7,32 @@ class PoolService(Service):
     @private
     async def format_disks(self, job, disks):
         """
-        Format all disks, putting all freebsd-zfs partitions created
-        into their respective vdevs and encrypting disks if specified for FreeBSD.
+        Format all disks, putting all ZFS partitions created  into their respective vdevs.
```
Comment (yocalebo) [inline src/middlewared/middlewared/plugins/pool_/format_disks.py:10]:
```suggestion
        Format all disks, putting all ZFS partitions created into their respective vdevs.
```

**Review by yocalebo — APPROVED** [review]
Confirmed on internal hardware that this works 😄

**Pull request merged**